lib/jsonapi/plugs/content_type_negotiation.ex

```diff
@@ -49,11 +49,18 @@ defmodule JSONAPI.ContentTypeNegotiation do
     {conn, content_type}
   end
 
+  defp respond({conn, @jsonapi, nil}) do
+    before_send(conn)
+  end
   defp respond({conn, @jsonapi, @jsonapi}) do
+    before_send(conn)
+  end
+  defp respond({conn, @jsonapi, _accepts}), do: send_error(conn, 406)
```

Comment:
I think one part this is missing is IF and ONLY IF "accepts" contains the json api media type, then throw 406.  Otherwise, if sent "application/json", don't throw 406.  Is this correct?

Reply:
Check out this PR https://github.com/emberjs/data/issues/3880